docs/arcadia-users-group/20221011-developer-productivity/lesson.md

```diff
@@ -0,0 +1,171 @@
+- It has tooling built-in to enable remote SSH access. So you can use an EC2 instance and navigate it through these tools.
+
+### Installing VS Code
+
+Download and install VS Code by going to [this URL](https://code.visualstudio.com/download). Once installed, turn on sync across devices and login with your GitHub account.
+
+## Customizing VS Code
+
+You can just use VS Code as is. This section suggests a couple base-level customizations to make your life a little easier.
+
+### Settings
+
+**Note:** There are hundreds of settings you can customize, we'll only focus on the essentials.
+
+Open up the settings panel by hitting Command + , or by selecting "Code" -> "Preferences" -> "Settings" from the top left corner. There are two ways to adjust settings:
+
+1. You can search through them using the search bar
+2. You can edit the underlying JSON blob that specifies the user settings.
+
+Today, we'll mostly do #1, but if you're interested in following the second method, a simplified version of all my settings can be found in [this file](config-and-dotfiles/vscode_settings.json). So, let's get started.
+
+- **Auto Save:** Search for "Auto Save" in the settings menu. And adjust it to the non-default value (which is none). Mine is "afterDelay".
+- **Font Size:** Search for "Font Size" and adjust it as you wish. I like 14pts font.
+- **Trim Trailing Whitespace:**: Search for it and turn it on, so your code doesn't have trailing spaces or tabs.
+- **Insert Final Newline:** Search for it and turn it on. This will prevent you from receiving PR review comments on [this](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).
+- **Trim Final Newlines:** Again, turn it on. This will make sure there's only a single newline at the end of each file.
+- insertFinalNewline
+- **Format on Save:** Search for "Format on Save" and turn it on. This will make sure, whatever code formatter you use formats the code upon saving it.
+- **Default Terminal application:** Search for "Osx Exex" and type "iTerm.app".
+- **Default Terminal profile:** Search for "Default Profile: Osx" and choose "bash".
+- **Adding code in the PATH:** Open the command palette with Command + Shift + P. Search for "code" and choose the "Install 'code' command in PATH". This will make it so that you can open any file/directory from the terminal with `code`.
+
+### Extensions
+
+Before we go through this section, let's quit VS Code. Renavigate to the correct demo folder and re-open with VS Code:
+
+```{bash}
+cd ~/Desktop/arcadia-computational-training
+code .
+```
+
+For this section, we'll only go through the installation instructions of a few VS Code extensions and give you a list of other extensions you can install on your own time. For downloading and installing extensions, click on the "Extensions" tab from the left tab, search for the name of the extension and hit "install". Here are the extensions:
+
+- Prettier: Customization with "Format Document With"
```

Comment:
I'll do demos of these, but haven't written detailed instructions for them, because I didn't have time. I think that's okay, but maybe I can write them out later

Reply:
ya i think this is fine :)

Reply:
👍🏻